### ISSUE TITLE
Poor error when block not available in the backend

### ISSUE THREAD

**Issue by touilleMan** [Scille/parsec-cloud#688]
With a block not present in the backend (e.g. the corresponding S3 bucket has been flushed by mistake) the core endup with an uncatched exception:

```
Traceback (most recent call last):
  File "C:\Program Files\Parsec\site-packages\parsec\core\mountpoint\winfsp_operations.py", line 32, in translate_error
    yield
  File "C:\Program Files\Parsec\site-packages\parsec\core\mountpoint\winfsp_operations.py", line 279, in read
    buffer = self.fs_access.fd_read(file_context.fd, length, offset)
  File "C:\Program Files\Parsec\site-packages\parsec\core\mountpoint\thread_fs_access.py", line 50, in fd_read
    return self._portal.run(self.workspace_fs.transactions.fd_read, fh, size, offset)
  File "C:\Program Files\Parsec\site-packages\trio\_threads.py", line 105, in run
    return self._do_it(self._run_cb, afn, *args)
  File "C:\Program Files\Parsec\site-packages\trio\_threads.py", line 86, in _do_it
    return q.get().unwrap()
  File "C:\Program Files\Parsec\site-packages\outcome\_sync.py", line 111, in unwrap
    raise captured_error
  File "C:\Program Files\Parsec\site-packages\trio\_threads.py", line 59, in unprotected_afn
    return await afn(*args)
  File "C:\Program Files\Parsec\site-packages\parsec\core\fs\workspacefs\file_transactions.py", line 197, in fd_read
    await self.remote_loader.load_blocks(missing)
  File "C:\Program Files\Parsec\site-packages\parsec\core\fs\remote_loader.py", line 162, in load_blocks
    await self.load_block(access)
  File "C:\Program Files\Parsec\site-packages\parsec\core\fs\remote_loader.py", line 197, in load_block
    raise FSError(f"Cannot download block: {exc}") from exc
parsec.core.fs.exceptions.FSError: Cannot download block: Backend error `timeout`: Error code `timeout` returned
```

This lead to a cryptic error in the software accessing the data (it would be much better to have a popup from the gui explaining the data is not available in the backend)

**Comment by vxgmichel** [comment]
Partially fixed by #724 and the exception refactoring. @touilleMan should we close this issue or do you want to fix exception handling on the backend side?

**Comment by touilleMan** [comment]
I think the error returned by the backend is fine, so close it ;-)
